skywalking/agent/__init__.py

```diff
@@ -73,6 +73,15 @@ def __fini():
 
 
 def start():
+    flag = False
+    try:
+        from gevent import monkey
+        flag = monkey.is_module_patched("socket")
+    except Exception as e:
+        logger.warning("check gevent error:%s", e)
+    if flag:
+        import grpc.experimental.gevent as grpc_gevent
+        grpc_gevent.init_gevent()
```

Comment:
Can you please explain why we add these lines in the agent? I remember these in the other thread but I suppose these should be in the targeted application, because the agent has nothing to do with `monkey` at all?

Reply:
Here is the context: https://github.com/apache/skywalking/issues/5263#issuecomment-678732774

Reply:
Another scene,I use 'grpc_gevent.init_gevent()',but  don't have gevent(monkey.patch_all()),I use threadpool,it will had an error occurred:
Traceback (most recent call last):

  File "src/gevent/event.py", line 159, in gevent._gevent_cevent.Event.wait

gevent.exceptions.InvalidThreadUseError: (<Hub '' at 0x7f2f917627c0 epoll pending=0 ref=3 fileno=8 resolver=<gevent.resolver.thread.Resolver at 0x7f2f8eb07700 pool=<ThreadPool at 0x7f2f7de7a2e0 tasks=0 size=1 maxsize=10 hub=<Hub at 0x7f2f917627c0 thread_ident=0x7f2f8c914700>>> threadpool=<ThreadPool at 0x7f2f7de7a2e0 tasks=0 size=1 maxsize=10 hub=<Hub at 0x7f2f917627c0 thread_ident=0x7f2f8c914700>> thread_ident=0x7f2f8c914700>, None, <greenlet.greenlet object at 0x7f2f9045a5c0>)

Traceback (most recent call last):

  File "src/gevent/event.py", line 159, in gevent._gevent_cevent.Event.wait

Exception ignored in: 'grpc._cython.cygrpc.run_loop'

Traceback (most recent call last):

  File "src/gevent/event.py", line 159, in gevent._gevent_cevent.Event.wait

  File "src/gevent/_abstract_linkable.py", line 425, in gevent._gevent_c_abstract_linkable.AbstractLinkable._wait

  File "src/gevent/_abstract_linkable.py", line 425, in gevent._gevent_c_abstract_linkable.AbstractLinkable._wait

  File "src/gevent/_abstract_linkable.py", line 425, in gevent._gevent_c_abstract_linkable.AbstractLinkable._wait

  File "src/gevent/_abstract_linkable.py", line 191, in gevent._gevent_c_abstract_linkable.AbstractLinkable._capture_hub

  File "src/gevent/_abstract_linkable.py", line 191, in gevent._gevent_c_abstract_linkable.AbstractLinkable._capture_hub

  File "src/gevent/_abstract_linkable.py", line 191, in gevent._gevent_c_abstract_linkable.AbstractLinkable._capture_hub

gevent.exceptions.InvalidThreadUseError: (<Hub '' at 0x7f2f917627c0 epoll pending=0 ref=3 fileno=8 resolver=<gevent.resolver.thread.Resolver at 0x7f2f8eb07700 pool=<ThreadPool at 0x7f2f7de7a2e0 tasks=0 size=1 maxsize=10 hub=<Hub at 0x7f2f917627c0 thread_ident=0x7f2f8c914700>>> threadpool=<ThreadPool at 0x7f2f7de7a2e0 tasks=0 size=1 maxsize=10 hub=<Hub at 0x7f2f917627c0 thread_ident=0x7f2f8c914700>> thread_ident=0x7f2f8c914700>, None, <greenlet.greenlet object at 0x7f2f9045a5c0>)

gevent.exceptions.InvalidThreadUseError: (<Hub '' at 0x7f2f917627c0 epoll pending=0 ref=3 fileno=8 resolver=<gevent.resolver.thread.Resolver at 0x7f2f8eb07700 pool=<ThreadPool at 0x7f2f7de7a2e0 tasks=0 size=1 maxsize=10 hub=<Hub at 0x7f2f917627c0 thread_ident=0x7f2f8c914700>>> threadpool=<ThreadPool at 0x7f2f7de7a2e0 tasks=0 size=1 maxsize=10 hub=<Hub at 0x7f2f917627c0 thread_ident=0x7f2f8c914700>> thread_ident=0x7f2f8c914700>, None, <greenlet.greenlet object at 0x7f2f9045a670>)

Exception ignored in: 'grpc._cython.cygrpc.run_loop'

Reply:
A code two or more kinds of worker mode

Reply:
OK I can get you point, sounds good to me, can you please slightly polish line 80-81 so that the warning logs are only printed when `monkey` really exists in users application, at the moment, it seems that when there is no `monkey` package installed, it prints the warning logs? I think that's confusing for users, right?

Reply:
you're right,I've tried to optimize it。